conf4j-extras/conf4j-extras-yaml/main/java/com/sabre/oss/conf4j/yaml/source/YamlConfigurationSource.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * MIT License
+ *
+ * Copyright 2017 Sabre GLBL Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sabre.oss.conf4j.yaml.source;
+
+import com.sabre.oss.conf4j.source.ConfigurationEntry;
+import com.sabre.oss.conf4j.source.IterableConfigurationSource;
+import com.sabre.oss.conf4j.source.MapIterable;
+import com.sabre.oss.conf4j.source.OptionalValue;
+import org.apache.commons.lang3.StringUtils;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.reader.UnicodeReader;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.io.UncheckedIOException;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static com.sabre.oss.conf4j.source.OptionalValue.absent;
+import static com.sabre.oss.conf4j.source.OptionalValue.present;
+import static java.util.Collections.singletonMap;
+import static java.util.Objects.requireNonNull;
+
+public class YamlConfigurationSource implements IterableConfigurationSource {
+    private final Map<String, String> properties;
+
+    public YamlConfigurationSource(InputStream inputStream) {
+        requireNonNull(inputStream, "inputStream cannot be null");
+
+        Reader reader = new UnicodeReader(inputStream);
+        this.properties = createProperties(reader);
```

Comment:
This line should be inside try to make sure input stream is always closed properly.